File: 30-Days-of-Code/Day 10 - Binary Numbers/Solution.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
    int n;
    int count=0;
    
    scanf("%d", &n);
    
    while (n != 0) {
        n = n & (n<<1);
        count++;
    }
    printf("%d", count);

    return 0;
}

File: 30-Days-of-Code/Day 10 - Binary Numbers/Solution.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())
    
    binary = str(bin(n))
    
    currentMax = 0
    count = 0
    
    for i in binary:
        if i == '1':
            count += 1
        else:
            currentMax = max(count, currentMax)
            count = 0
            
    print(max(count, currentMax))
